pkg/api/util/db_test.go

```diff
@@ -682,3 +682,160 @@ func TestDeleteFileShareAclDBEntry(t *testing.T) {
+		in.ProfileId = ""
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetFileShare", context.NewAdminContext(), in.FileShareId).Return(&SampleFileShares[0], nil)
+		mockClient.On("CreateFileShareAcl", context.NewAdminContext(), in).Return(&SampleFileSharesAcl[2], nil)
+		db.C = mockClient
+
+		_, err := CreateFileShareAclDBEntry(context.NewAdminContext(), in)
+		expectedError := "profile id can not be empty when creating fileshare acl in db!"
+		assertTestResult(t, err.Error(), expectedError)
+	})
+
+	t.Run("Invalid Access Type", func(t *testing.T) {
+		in.ProfileId, in.Type = "d2975ebe-d82c-430f-b28e-f373746a71ca", "system"
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetFileShare", context.NewAdminContext(), in.FileShareId).Return(&SampleFileShares[0], nil)
+		mockClient.On("CreateFileShareAcl", context.NewAdminContext(), in).Return(&SampleFileSharesAcl[2], nil)
+		db.C = mockClient
+
+		_, err := CreateFileShareAclDBEntry(context.NewAdminContext(), in)
+		expectedError := fmt.Sprintf("invalid fileshare type: %v. Supported type is: ip", in.Type)
+		assertTestResult(t, err.Error(), expectedError)
+	})
+
+	t.Run("Empty Access To", func(t *testing.T) {
+		in.ProfileId, in.Type, in.AccessTo = "d2975ebe-d82c-430f-b28e-f373746a71ca", "ip", ""
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetFileShare", context.NewAdminContext(), in.FileShareId).Return(&SampleFileShares[0], nil)
+		mockClient.On("CreateFileShareAcl", context.NewAdminContext(), in).Return(&SampleFileSharesAcl[2], nil)
+		db.C = mockClient
+
+		_, err := CreateFileShareAclDBEntry(context.NewAdminContext(), in)
+		expectedError := "accessTo is empty. Please give valid ip segment"
+		assertTestResult(t, err.Error(), expectedError)
+	})
+
+	t.Run("Invalid Ip Segment", func(t *testing.T) {
+		in.ProfileId, in.Type, in.AccessTo = "d2975ebe-d82c-430f-b28e-f373746a71ca", "ip", "201.100.101.8/9.9"
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetFileShare", context.NewAdminContext(), in.FileShareId).Return(&SampleFileShares[0], nil)
+		mockClient.On("CreateFileShareAcl", context.NewAdminContext(), in).Return(&SampleFileSharesAcl[2], nil)
+		db.C = mockClient
+
+		_, err := CreateFileShareAclDBEntry(context.NewAdminContext(), in)
+		expectedError := fmt.Sprintf("invalid IP segment %v", in.AccessTo)
+		assertTestResult(t, err.Error(), expectedError)
+	})
+
+	t.Run("Invalid Ip", func(t *testing.T) {
+		in.ProfileId, in.Type, in.AccessTo = "d2975ebe-d82c-430f-b28e-f373746a71ca", "ip", "201.100.101"
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetFileShare", context.NewAdminContext(), in.FileShareId).Return(&SampleFileShares[0], nil)
+		mockClient.On("CreateFileShareAcl", context.NewAdminContext(), in).Return(&SampleFileSharesAcl[2], nil)
+		db.C = mockClient
+
+		_, err := CreateFileShareAclDBEntry(context.NewAdminContext(), in)
+		expectedError := fmt.Sprintf("%v is not a valid ip. Please give the proper ip", in.AccessTo)
```

Comment:
Here, the error is too generic. typically, if i use this API with multiple IPs, what do I make of the error message ? Which IP is invalid ? Is the share created for the remaining valid IPs OR not created for any IP ? With what new input do i remedy the situation ? Please fix.

Reply:
Valid IP format is known for user right, IP format is common for all, 3 dots separated with 4 numbers. When it comes to format standard only IPv4 is supported across the opensds.